torchdata/datapipes/iter/util/combining.py

```diff
@@ -213,3 +216,139 @@ def __iter__(self) -> Iterator:
 
     def __len__(self) -> int:
         return len(self.source_iterdatapipe)
+
+
+def _drop_index(idx_data):
+    idx, data = idx_data
+    return data
+
+
+@functional_datapipe("round_robin_demux")
+class RoundRobinDemultiplexerIterDataPipe(IterDataPipe):
+    r"""
+    Splits the input DataPipe into multiple child DataPipes in the round-robin order (functional name: ``round_robin_demux``).
+    A list of the child DataPipes is returned from this operation.
+
+    Args:
+        datapipe: Iterable DataPipe being filtered
+        num_instances: number of instances of the DataPipe to create
+        buffer_size: this defines the maximum number of inputs that the buffer can hold across all child
+            DataPipes while waiting for their values to be yielded.
+            Defaults to ``1000``. Use ``-1`` for the unlimited buffer.
+
+    Examples:
+        >>> from torchdata.datapipes.iter import IterableWrapper
+        >>> def odd_or_even(n):
+        ...     return n % 2
+        >>> source_dp = IterableWrapper(range(5))
+        >>> dp1, dp2 = source_dp.round_robin_demux(2)
+        >>> list(dp1)
+        [0, 2, 4]
+        >>> list(dp2)
+        [1, 3]
+    """
+
+    def __new__(cls, datapipe: IterDataPipe, num_instances: int, buffer_size: int = 1000):
+        if num_instances < 1:
+            raise ValueError(f"Expected `num_instaces` larger than 0, but {num_instances} is found")
+        if num_instances == 1:
+            warnings.warn(
+                "The operation of `round_robin_demux` with `num_instances=1` is an no-op and returns the provided `datapipe` directly"
+            )
+            return datapipe
+
+        datapipe = datapipe.enumerate()
+        container = _RoundRobinDemultiplexerIterDataPipe(datapipe, num_instances, buffer_size=buffer_size)
+        return [_ChildDataPipe(container, i).map(_drop_index) for i in range(num_instances)]
```

Comment:
As a Python noob, TIL I learnt `X()` in Python may not return a instance of `X`

https://www.geeksforgeeks.org/__new__-in-python/

> ...  because other class constructors can be called by __new__ method or it can simply return other objects as an instance of this class.

Will `_ChildDataPipe.__init__` be called for each instance in the list? ~